src/vidpf.rs

```diff
@@ -528,11 +528,17 @@ impl<W: VidpfValue> Encode for VidpfPublicShare<W> {
     }
 
     fn encoded_len(&self) -> Option<usize> {
-        let control_bits_count = self.cw.len() * 2;
-        let mut len = (control_bits_count + 7) / 8 + self.cw.len() * 48;
-        for cw in self.cw.iter() {
-            len += cw.weight.encoded_len()?;
-        }
+        // We assume the weight has the same length at each level of the tree.
+        let weight_parameter = self
```

Comment:
nit: This name suggests a relationship to the `ValueParameter` associated type, which counts field elements, but this counts bytes instead. How about `weight_len` or `weight_size`?

Reply:
Done. We might also consider doing away with the `WeightParameter` associated type in the future. Want to do that here, or can it wait?

Reply:
Are you thinking of hardcoding `ValueParameter` to be a `usize`? I think the flexibility still might be worth it. For example, you may want to generate a tensor of field elements instead of just a vector. It could also be useful for switching between two field sizes, like in Poplar1, but without relying on the aggregation parameter as much.